### PR TITLE
Improve PR template

### PR DESCRIPTION
**Pull request by lukehinds**
Serveral changes are implemented here.

* Request that an issue first be raised
* Ask for more details on how to test
* Remove references to mattermost
* Add documentation section

Signed-off-by: Luke Hinds <luke@redhat.com>

#### Ticket Link

Fixes

#### Release Note

```release-note

```

### CODE REVIEW

**Comment by lukehinds** [comment]
@cpanato

**Review by cpanato — DISMISSED** [review]
thanks for the updates lgtm

File: pull_request_template.md [hunk 1]
```diff
@@ -1,8 +1,9 @@
 <!--
-Thanks for opening a pull request!
+Thanks for opening a pull request! Please do not just delete this text.  The three fields below are mandatory.
 
 Please remember to:
-- mention any issue(s) that this PR closes using a closing keyword as well as the issue number, such as "Closes #XYZ" or "Resolves sigstore/repo-name#XYZ", cf.
+- This PR requires an isssue. If it is a new feature, the issue should proceed the PR and will have allowed sufficent time for discussions to take place. Pleases use
```
Comment (mbestavros) [inline pull_request_template.md:5]: "isssue" -> "issue"

**Review by mbestavros — CHANGES_REQUESTED** [review]
Looks good to me, apart from a small typo.

**Review by mbestavros — APPROVED** [review]

**Pull request merged**